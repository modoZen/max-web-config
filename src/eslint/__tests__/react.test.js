const { getEslintReactConfig } = require('../react');

describe('Eslint tests', () => {
  it('Should return eslint config values', () => {
    expect(getEslintReactConfig()).toMatchSnapshot();
  });
});
