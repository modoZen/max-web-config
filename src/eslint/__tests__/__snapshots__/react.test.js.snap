// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eslint tests Should return eslint config values 1`] = `
{
  "env": {
    "browser": true,
    "es2021": true,
    "jest": true,
  },
  "extends": [
    "airbnb",
    "eslint:recommended",
    "plugin:react/recommended",
    "plugin:react-hooks/recommended",
    "plugin:react/jsx-runtime",
    "plugin:@typescript-eslint/recommended",
    "plugin:prettier/recommended",
  ],
  "overrides": [
    {
      "files": [
        "*.stub.ts",
        "*.mock.ts",
        "*.test.ts",
      ],
      "rules": {
        "@typescript-eslint/explicit-function-return-type": 0,
        "@typescript-eslint/no-empty-function": 0,
        "no-underscore-dangle": 0,
      },
    },
    {
      "files": [
        "**/*.js",
      ],
      "rules": {
        "@typescript-eslint/explicit-function-return-type": 0,
        "@typescript-eslint/no-var-requires": 0,
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": "latest",
    "sourceType": "module",
  },
  "plugins": [
    "@typescript-eslint",
    "react",
    "react-hooks",
  ],
  "rules": {
    "@typescript-eslint/consistent-type-definitions": 2,
    "@typescript-eslint/explicit-function-return-type": [
      2,
      {
        "allowExpressions": true,
      },
    ],
    "@typescript-eslint/no-explicit-any": 0,
    "@typescript-eslint/no-shadow": 2,
    "@typescript-eslint/no-unused-vars": [
      2,
      {
        "ignoreRestSiblings": true,
      },
    ],
    "@typescript-eslint/no-use-before-define": [
      2,
      {
        "classes": false,
        "variables": false,
      },
    ],
    "camelcase": [
      2,
      {
        "ignoreDestructuring": true,
        "properties": "never",
      },
    ],
    "curly": 2,
    "default-case": 0,
    "dot-notation": 0,
    "import/extensions": 0,
    "import/no-extraneous-dependencies": [
      2,
      {
        "devDependencies": true,
      },
    ],
    "import/no-unresolved": 0,
    "import/prefer-default-export": 0,
    "no-console": 2,
    "no-empty": [
      2,
      {
        "allowEmptyCatch": true,
      },
    ],
    "no-empty-function": 0,
    "no-shadow": 0,
    "no-underscore-dangle": [
      2,
      {
        "allowAfterThis": true,
      },
    ],
    "no-unused-expressions": [
      2,
      {
        "allowTernary": true,
      },
    ],
    "no-use-before-define": 0,
    "prefer-destructuring": 0,
    "react-hooks/exhaustive-deps": 2,
    "react-hooks/rules-of-hooks": 2,
    "react/function-component-definition": [
      2,
      {
        "namedComponents": "arrow-function",
      },
    ],
    "react/jsx-filename-extension": [
      1,
      {
        "extensions": [
          ".js",
          ".jsx",
          ".ts",
          ".tsx",
        ],
      },
    ],
    "react/jsx-no-useless-fragment": [
      2,
      {
        "allowExpressions": true,
      },
    ],
    "react/jsx-props-no-spreading": [
      1,
      {
        "custom": "ignore",
        "explicitSpread": "enforce",
        "html": "enforce",
      },
    ],
    "react/require-default-props": 0,
  },
  "settings": {
    "import/extensions": [
      ".js",
      ".jsx",
      ".ts",
      ".tsx",
    ],
    "import/parsers": {
      "@typescript-eslint/parser": [
        ".ts",
        ".tsx",
      ],
    },
    "import/resolver": {
      "node": {
        "extensions": [
          ".js",
          ".jsx",
          ".ts",
          ".tsx",
        ],
        "paths": [
          "src",
        ],
      },
    },
    "react": {
      "version": "detect",
    },
  },
}
`;
